### ISSUE TITLE
wrong article workflow for InstantILL

### ISSUE THREAD

**Issue by JosephMcArthur** [OAButton/discussion#1135]
TODO Joe to Expand

**Comment by JosephMcArthur** [comment]
This workflow is used when:
* 1 We can't find any match
* 2 We get the wrong match

For 1, people should enter the workflow after an unsuccessful search. For 2, people will enter the workflow on a 'matched' page by clicking a link/button.

After that, the behavior can be the same though. 

* Whatever we do have (be it just a title, or more) should be displayed in the relavent boxes
* the user should be show fields to fill out
* some of those fields should be required

The design, fields, can be seen in V13 of the mock-ups. 

After the fields are filled in, we should once again do an availability and subscription search.

**Comment by JosephMcArthur** [comment]
- [ ] add actual required check

then close

**Comment by JosephMcArthur** [comment]
You're probably aware of this, but currently this doesn't submit with or without things in the field.

**Comment by JosephMcArthur** [comment]
and lets have 'DOI or URL' be 'Article DOI or URL'

**Comment by JosephMcArthur** [comment]
- [ ] let's have 'Can you please tell us the article details?' be 'Sorry that didn't work! Can you please tell us the article details?'

**Comment by markmacgillivray** [comment]
Above changes have been made and continue button does now work, I was just still writing that required check code when you last tried it. So this is ready to test now.

**Comment by JosephMcArthur** [comment]
Haven't yet got this to work. 

When I managed to get further than the form, on submission I got this: 

![image](https://user-images.githubusercontent.com/4953703/55552475-596f4d00-56d5-11e9-8275-d240dd010ccf.png)

Not clear why that error comes up, and why it says what it says. 

The date section of these appears to have accepted a two letter date too. 

It would be great if you could generally test this more, seems super buggy.

**Comment by JosephMcArthur** [comment]
Another thing, when the title was long enough to pass the first check, but not get a match, it seems I got sent to a blank page with just the option to say 'this is not what I searched'. I think we should just land them on the form.

**Comment by markmacgillivray** [comment]
What do you mean by date section? The year field? Do you want it to only accept 4 digits?

When you say we should just land them on the form, that is not possible when we do not have a form to land them on. Do you mean that you want the wrong article workflow to actually end with the appropriate "complete request" button, and not a "continue" button at all, so that when they put in those extra metadata details we just immediately create an ILL the appropriate way? At the moment you could end up back on a page showing nothing but the "this is not what I searched" because we do another round of looking for availability and then we are showing them that page again - but if we have not found anything suitable to match their newly provided metadata, then we have nothing to show there.

**Comment by JosephMcArthur** [comment]
> What do you mean by date section? The year field? Do you want it to only accept 4 digits?

Yes, the year field. And yes, I think only taking 4 digits makes sense. 

> Another thing, when the title was long enough to pass the first check, but not get a match, it seems I got sent to a blank page with just the option to say 'this is not what I searched'. I think we should just land them on the form.

Let me clarify the above, 'the form' in this case is the wrong article form, not the campuses ILL form.
What I want to avoid is someone searching something, it not finding anything and landing them on a blank page where their only option is to click through to the wrong article workflow. We should just put them right onto the wrong article form.

**Comment by markmacgillivray** [comment]
OK, year field must be 4 digits long, and if no useful result is found the user is sent straight to the wrong article form.

**Comment by JosephMcArthur** [comment]
I just did this putting 'something' in every box. When I searched, I got back: 

> something
> 2019, Lancet (London, England)

as the article. 

This shouldn't be possible, as it suggests we overwrote 'something' the journal with new info. 

With this workflow, the system should do a new oa look up, and new subs (soon) look up, and try to *add* meta-data, but not overwrite data. So, I shouldn't have been able to get the above result.
